receiver/prometheusreceiver/metrics_receiver.go

```diff
@@ -92,36 +114,48 @@ func (r *pReceiver) Start(_ context.Context, host component.Host) error {
 
 	allocConf := r.cfg.TargetAllocator
 	if allocConf != nil {
-		go func() {
-			// immediately sync jobs and not wait for the first tick
-			savedHash, _ := r.syncTargetAllocator(uint64(0), allocConf, baseCfg)
-			r.targetAllocatorIntervalTicker = time.NewTicker(allocConf.Interval)
-			for {
-				<-r.targetAllocatorIntervalTicker.C
-				hash, err := r.syncTargetAllocator(savedHash, allocConf, baseCfg)
-				if err != nil {
-					r.settings.Logger.Error(err.Error())
-					continue
-				}
-				savedHash = hash
-			}
-		}()
+		r.initTargetAllocator(allocConf, baseCfg)
 	}
 
+	r.reloadReady.Close()
+
 	return nil
 }
 
+func (r *pReceiver) initTargetAllocator(allocConf *targetAllocator, baseCfg *config.Config) {
+	r.settings.Logger.Info("Starting target allocator discovery")
+	// immediately sync jobs and not wait for the first tick
+	savedHash, err := r.syncTargetAllocator(uint64(0), allocConf, baseCfg)
+	if err != nil {
+		r.settings.Logger.Error(err.Error())
+		return
```

Comment:
Should `initTargetAllocator()` return an error when this fails?  Previously, the go routine would continue to try calling `syncTargetAllocator()`, even if an error occurred the first time.